applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, null, getETagValue(ETAG_ANY));
     }
 
     /**
-     * Gets the ConfigurationSetting object for the specified key and KeyValueFilter2 parameters.
-     * @param key the key being retrieved
-     * @param filter options for the request
-     * @return ConfigurationSetting object
+     * Creates or updates a configuration value in the service.
+     *
+     * <p>
+     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
+     * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
+     * updated.
+     * </p>
+     *
+     * @param configurationSetting The configuration setting to create or update.
+     * @return ConfigurationSetting that was created or updated.
+     * @throws com.azure.common.http.rest.RestException If the {@link ConfigurationSetting#etag()} was specified, is not
+     *                                               {@link ConfigurationClient#ETAG_ANY}, and the current configuration
+     *                                               value's etag does not match.
      */
-    public Mono<RestResponse<ConfigurationSetting>> getKeyValue(String key, KeyValueFilter filter) {
-        if (key == null || key.isEmpty()) {
-            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
-        }
-        if (filter != null) {
-            return service.getKeyValue(baseUri.toString(), key, filter.label(), filter.fields(),
-                    filter.acceptDateTime(), filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.getKeyValue(baseUri.toString(), key, null, null, null, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting) {
+        Validator.validate(configurationSetting);
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
+                .withValue(configurationSetting.value())
+                .withContentType(configurationSetting.contentType())
+                .withTags(configurationSetting.tags());
+
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, getETagValue(configurationSetting.etag()), null);
     }
 
+
     /**
-     * Deletes the ConfigurationSetting.
-     * @param key keyValue to delete
-     * @return the deleted ConfigurationSetting or none if didn't exist.
+     * Updates an existing configuration value in the service. The setting must already exist.
+     *
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     * </p>
```

Comment:
This is also true for `set`. Should it be added to the doc part?

Reply:
Added